plugins/modules/adguardhome_rewrite.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2025 Markus Bergholz
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+module: adguardhome_rewrite
+
+short_description: Add, update or delete DNS rewrite rules from AdGuard Home
+
+version_added: 3.3.0
+
+description:
+  - Add, update or delete DNS rewrite rules from AdGuard Home.
+extends_documentation_fragment:
+  - community.dns.adguardhome.connectivity
+options:
+  state:
+    description:
+      - wether a rewrite rule should be added/updated O(state=present) or removed O(state=absent).
+    type: str
+    default: present
+    choices:
+      - present
+      - absent
+  domain:
+    description:
+      - domain or wildcard domain that you want to be rewritten by AdGuard Home.
+    type: str
+    required: true
+  answer:
+    description:
+      - value for the domain rewrite.
+      - required when O(state=present).
+      - value can be a CNAME, A or AAAA record.
+    type: str
+    required: false
+attributes:
+  check_mode:
+    support: full
+    description: Can run in C(check_mode) and return changed status prediction without modifying target.
```

Comment:
The doc fragment is there for not having to repeat the boilerplate text everywhere.
```suggestion
```